### ISSUE TITLE
Getting value from map[string]interface{} panics

### ISSUE THREAD

**Issue by Albibek** [containous/yaegi#312]
The following program `sample.go` triggers a panic:

```go
package main

import "github.com/containous/yaegi/interp"

const src = `package foo

func use(interface{}){
}
func Bar() { 
m:= make(map[string]interface{})
m["a"] = 5
z := m["a"]
use(z)
}`

func main() {
	i := interp.New(interp.Options{})

	_, err := i.Eval(src)
	if err != nil {
		panic(err)
	}

	v, err := i.Eval("foo.Bar")
	if err != nil {
		panic(err)
	}

	bar := v.Interface().(func())

	bar()
}
```

Expected result:
```console
nothing
```

Got:
```console
6:5: panic
panic: reflect.Set: value of type interface {} is not assignable to type interp.valueInterface [recovered]
	panic: reflect.Set: value of type interface {} is not assignable to type interp.valueInterface

goroutine 1 [running]:
github.com/containous/yaegi/interp.runCfg.func1(0x8303c0, 0x846d00)
	/tmp/gopath000332052/pkg/mod/github.com/containous/yaegi@v0.0.4/interp/run.go:107 +0x220
panic(0x23a940, 0x80c708)
	/usr/local/go/src/runtime/panic.go:522 +0x240
reflect.Value.assignTo(0x23a3c0, 0x80c700, 0x94, 0x2672a5, 0xb, 0x242b00, 0x0, 0x80c408, 0x23a940, 0x80c450, ...)
	/usr/local/go/src/reflect/value.go:2339 +0x420
reflect.Value.Set(0x242b00, 0x80a4a0, 0x199, 0x23a3c0, 0x80c700, 0x94)
	/usr/local/go/src/reflect/value.go:1473 +0xc0
github.com/containous/yaegi/interp.getIndexMap.func2(0x8303c0, 0x2879b4, 0x830330, 0x846d00)
	/tmp/gopath000332052/pkg/mod/github.com/containous/yaegi@v0.0.4/interp/run.go:955 +0x140
github.com/containous/yaegi/interp.runCfg(0x846d00, 0x8303c0)
	/tmp/gopath000332052/pkg/mod/github.com/containous/yaegi@v0.0.4/interp/run.go:112 +0x80
github.com/containous/yaegi/interp.genFunctionWrapper.func1.1(0x3d773c, 0x0, 0x0, 0x80a460, 0x3d773c, 0x24ffa0, 0xd3a60, 0x0)
	/tmp/gopath000332052/pkg/mod/github.com/containous/yaegi@v0.0.4/interp/run.go:470 +0x400
reflect.callReflect(0x83ec60, 0x862f80, 0x862f74, 0x0)
	/usr/local/go/src/reflect/value.go:536 +0x3a0
reflect.makeFuncStub(0x231fe0, 0x83ec60, 0x13, 0x5401, 0x231fe0, 0x83ec60, 0x13, 0x0, 0x0, 0x54f3, ...)
	/usr/local/go/src/reflect/asm_amd64p32.s:20 +0x40
main.main()
	/tmp/sandbox225636937/prog.go:31 +0x120
```

[Link to playground](https://play.golang.org/p/2Vk8-ku1W_k)

**Comment by bysans** [comment]
I'm stuck on this bug too. This bug has to fixed to use json types.

**Comment by hrsh7th** [comment]
Thanks providing `yaegi` for all yaegi developer. I love this package.

I have similar problem.
Minimal problem sample is here.

```go
package main

import (
	"fmt"

	"github.com/containous/yaegi/interp"
	"github.com/containous/yaegi/stdlib"
)

func main() {
	i := interp.New(interp.Options{})
	i.Use(stdlib.Symbols)
	i.Eval(`
    package main

    import "fmt"
    import "encoding/json"

    func Run(jsonbytes []byte) {
      var params map[string]interface{}
      if err := json.Unmarshal(jsonbytes, &params); err != nil {
        fmt.Println("marshal failed.")
      }

      fmt.Printf("marshal success: %s", params["property"].(string))
    }
  `)
	v, err := i.Eval("Run")
	if err != nil {
		fmt.Println("Can't get `Run`")
		return
	}

	run := v.Interface().(func([]byte))

	run([]byte("{\"property\":\"test\"}"))
}
```

`json.Unmarshal` is success but seems panic on accessing json property.

**Comment by nrwiersma** [comment]
This seems to be fixed in `v0.8.11`. Can you confirm please?

**Comment by Albibek** [comment]
Yes, It now works as intended, thank you.
